src/landlab/components/overland_flow/generate_overland_flow_implicit_kinwave.py

```diff
@@ -328,12 +345,23 @@ def run_one_step(self, dt):
             #
             #   $\alpha = \frac{\Sigma W S^{1/2} \Delta t}{A C_r}$
             cores = self._grid.core_nodes
-            self._alpha[cores] = (
-                self._vel_coef
-                * self._grad_width_sum[cores]
-                * dt
-                / (self._grid.area_of_cell[self._grid.cell_at_node[cores]])
-            )
+
+            # Calculate alpha; try for roughness as a float, else as array of floats
+            if not hasattr(self._roughness, "__len__"):
+                self._alpha[cores] = (
+                    self._vel_coef
+                    * self._grad_width_sum[cores]
+                    * dt
+                    / (self._grid.area_of_cell[self._grid.cell_at_node[cores]])
+                )
+            elif hasattr(self._roughness, "__len__"):
+                # if manning's n in a field
+                self._alpha[cores] = (
+                    self._vel_coef[cores]
+                    * self._grad_width_sum[cores]
+                    * dt
+                    / (self._grid.area_of_cell[self._grid.cell_at_node[cores]])
+                )
```

Comment:
To avoid repeated code and, I think, make this a little clearer, how about,
```python
if np.isscalar(self._roughness):
    roughness_at_core_nodes = self._roughness
else:
    roughness_at_core_nodes = self._roughness[cores]

self._alpha[cores] = (
    self._grad_width_sum[cores]
    * dt
    / (self._grid.area_of_cell[self._grid.cell_at_node[cores]])
    / roughness_at_core_nodes
)
```